backend/path-project.go

```diff
@@ -141,3 +156,34 @@ func handleProjectUpdate(ctx context.Context, req *logical.Request, data *framew
 		Data: item.Data(),
 	}, nil
 }
+
+func handleProjectDelete(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	projectName := data.Get("project").(string)
+	project, err := loadProject(ctx, req.Storage, projectName)
+	if err != nil {
+		return nil, err
+	}
+
+	if project == nil {
+		return logical.ErrorResponse("project %s is not configured in vault", projectName), nil
+	}
+
+	// Remove project name from vault
+	err = req.Storage.Delete(ctx, KeyProjectConfigPrefix+projectName)
+	if err != nil {
+		return nil, err
+	}
+
+	// Cleanup project dsn entries from vault too
+	err = req.Storage.Delete(ctx, KeyDsnPrefix+projectName+"/*")
```

Comment:
this is not correct. pattern matching does not work in storage operations. you need to list everything under the path prefix using `storage.List()` and then call delete on each item.

Reply:
Done